MyPerfectOnboardingApplication/Sources/MyPerfectOnboarding.Contracts/Dependency/IContainer.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace MyPerfectOnboarding.Contracts.Dependency
+{
+    public interface IContainer : IDisposable
+    {
+        IContainer Register<TIType>(TIType instance);
```

Comment:
```suggestion
        IContainer Register<TInstance>(TInstance instance);
```
(the `Type` is already there - that is what the `T` means 😛)